main.go

```diff
@@ -33,25 +34,46 @@ func main() {
 }
 
 func getToken() (string, error) {
```

Comment:
Let's split this out into its own helper file in a top level package. i dropped more detailed comments on it in a DM on Slack. But I think this is the first helper for the new API we'll build for this package.

Once that's done we can add some basic unit tests for it. If you're comfortable with Go testing, you're welcome to take that on. Otherwise, I can do it in a follow up PR to give you some examples to reference.

We can either drop main.go or move into a cmd/ subdirectory if we want to keep it for initial testing purposes.